src/magicbox/ResultPreviewsManager.ts

```diff
@@ -173,12 +187,18 @@ export class ResultPreviewsManager {
     ).el;
   }
 
+  private getExternalOptions() {
```

Comment:
Hmm, I am a little concerned about needing an event to retrieve the options. Could you share a diagram of the relevant classes and inter-play between right now? e.g. a google doc. Having a higher-level picture would help me contextualise the problem and give a better review,

Reply:
The alternative I can think of is that this first event could be removed and instead of receiving promises it could receive callbacks that return a promise instead.

Here's a diagram to help contextualize the interactions between `ResultPreviewsManager` and `QuerySuggestPreview`:

![ResultPreviewsManager and QuerySuggestPreview](https://user-images.githubusercontent.com/54454747/69348401-1eaa2d00-0c44-11ea-828a-2473a330db44.png)

I can update or remake this diagram if needed.

Reply:
Thanks for the diagram. My first thoughts are whether we could simplify by:

1. Inverting QuerySuggestPreview (QSP) and ResultPreviewsManager (RPM) such that QSP becomes the root/entry since it is the component specified in the mark-up. That allows us to pass the options from QSP to RPM avoiding the need to trigger an event to collect the info.

2. Extracting the server query/template-building logic out of QSP into either RPM or a new dedicated class to avoid needing to back-communicate.

One remaining question is where the update call is coming from. Is it the QSP?

For this PR, let's keep what we have. I can transition over to the search-ui next week to help with the feature, and verify if the above suggestions make sense.

Reply:
The update call is coming from `SuggestionsManager`, which receives it from user input (selecting a suggestion). RPM is initialized by SuggestionsManager.

It's definitely worth checking out how this could be simplified, I'll merge this PR and I'll create an issue about this.